src/main/java/com/mraof/minestuck/client/gui/ReadableSburbCodeScreen.java

```diff
@@ -0,0 +1,293 @@
+package com.mraof.minestuck.client.gui;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.mraof.minestuck.Minestuck;
+import com.mraof.minestuck.util.MSTags;
+import net.minecraft.client.gui.chat.NarratorChatListener;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.gui.screens.inventory.PageButton;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.Style;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.resources.Resource;
+import net.minecraft.world.level.block.Block;
+import org.apache.commons.lang3.mutable.MutableInt;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Code is modified components of, or inspired by, WinGameScreen/ReadBookScreen/StoneTabletScreen.
+ * It reads the nucleotide sequence stored in rana_temporaria_sec22b.txt and associates a number of lines of the text to each hieroglyph from the block tag with one extra pseudo-hieroglyph in the form of the paradox code
+ */
+public class ReadableSburbCodeScreen extends Screen
```

Comment:
I suspect that more fields here should be private. What is your opinion?

Reply:
That seems reasonable